filter_test.go

```diff
@@ -1524,6 +1527,7 @@ func TestFilterFlowerAddDel(t *testing.T) {
 		EncSrcIPMask:  testMask,
 		EncDestPort:   8472,
 		EncKeyId:      1234,
+		IPProto:       ipproto,
```

Comment:
I do not see this function checking this parameter once it reads the added filter from kernel.

Reply:
The check is added to #1594